### ISSUE TITLE
Support more targets

### ISSUE THREAD

**Issue by taiki-e** [taiki-e/setup-cross-toolchain-action#2]
[rust-cross-toolchain](https://github.com/taiki-e/rust-cross-toolchain), on which this project is based, supports so many targets.

**Comment by zeeshanlakhani** [comment]
Hey @taiki-e,

I was going to add a PR to `cargo-hack` for it to build a release for illumos (x86_64-unknown-illumos). I see that rust-cross-toolchain already supports this, so it should be straightforward to update this action and easily run it for `cargo-hack`, right? I just wanted to ensure I wasn't missing something before doing the work. 

Other options: 
- We can use `cross` directly in another job or via a conditional, but I'm guessing you'd prefer the toolchain set up already?

**Comment by taiki-e** [comment]
>  I see that rust-cross-toolchain already supports this, so it should be straightforward to update this action and easily run it for cargo-hack, right?

Yeah, I guess it would generally be the same as the NetBSD case, except that it does not support multiple OS versions.

Probably could be implemented by simply adding an `| *-illumos*` to the two lines in main.sh, and updating the CI and readme.

https://github.com/taiki-e/setup-cross-toolchain-action/blob/5512f65877f179333be1718ac2acbdb263e8e610/main.sh#L501
https://github.com/taiki-e/setup-cross-toolchain-action/blob/5512f65877f179333be1718ac2acbdb263e8e610/main.sh#L657

https://github.com/taiki-e/setup-cross-toolchain-action/blob/5512f65877f179333be1718ac2acbdb263e8e610/.github/workflows/ci.yml#L152-L155
